internal/services/eventhub/eventhub_namespace_customer_managed_key_resource.go

```diff
@@ -208,8 +256,31 @@ func expandEventHubNamespaceKeyVaultKeyIds(input []interface{}) (*[]namespaces.K
 	return &results, nil
 }
 
-func flattenEventHubNamespaceKeyVaultKeyIds(input *namespaces.Encryption) ([]interface{}, error) {
-	results := make([]interface{}, 0)
+// Get the user managed id for the custom keys
+func getUserManagedIdentity(input *namespaces.Encryption) *[]interface{} {
+	if input == nil || input.KeyVaultProperties == nil {
+		return nil
+	}
+
+	// we can only have a single user managed id for N number of keys, azure portal only allows setting a single one and then applies it to each key
+	for _, item := range *input.KeyVaultProperties {
+		if item.Identity != nil {
+			return &[]interface{}{
+				map[string]interface{}{
+					"type":         "UserAssigned",
+					"identity_ids": []string{*item.Identity.UserAssignedIdentity},
+					"principal_id": "",
+					"tenant_id":    "",
+				},
+			}
+		}
+	}
```

Comment:
we've got consistent Expand and Flatten functions we use across the Provider for this - we're using the Expand above, so can we update [this to use the Flatten method here](https://github.com/hashicorp/go-azure-helpers/blob/97adae4d709cc27376d863663e2ac6a775bbcfd6/resourcemanager/identity/system_or_user_assigned_list.go#L82-L110) so that this behaviour is normalised across the provider?

Reply:
So i dug into this a bit, and the reason I didn't use the existing Flatten method is because the data type that is returned for the `encryption.keyVaultProperties.identity` doesn't match the existing `SystemOrUserAssignedList` type. The identity block declared in this resource was used since it already existed, but the Identity data that is returned is just the ID of the User Managed ID that is associated with the encryption, alongside the Key Vault details.

Comparison of payloads:
```json
{
    "sku": {
        "name": "Premium",
        "tier": "Premium",
        "capacity": 1
    },
    "id": "/subscriptions/c6d2f82d-4baf-4b80-8d69-1fc6fbf92ad8/resourceGroups/bharrison-Tz6Ium/providers/Microsoft.EventHub/namespaces/bharrison-namespace-Tz6Ium",
    "name": "bharrison-namespace-Tz6Ium",
    "type": "Microsoft.EventHub/Namespaces",
    "location": "East US",
    "tags": {},
    "identity": {  <-- Parent Event Hub Identity Block (standard shape)
        "type": "UserAssigned",
        "userAssignedIdentities": {
            "/subscriptions/c6d2f82d-4baf-4b80-8d69-1fc6fbf92ad8/resourceGroups/bharrison-Tz6Ium/providers/Microsoft.ManagedIdentity/userAssignedIdentities/test": {
                "clientId": "5d7f143c-610c-4342-a574-941237de2d13",
                "principalId": "5fe787f4-e013-4202-8ef7-70e03c79b574"
            }
        }
    },
    "properties": {
        "minimumTlsVersion": "1.2",
        "publicNetworkAccess": "Enabled",
        "disableLocalAuth": false,
        "zoneRedundant": true,
        "isAutoInflateEnabled": false,
        "maximumThroughputUnits": 0,
        "encryption": {
            "keySource": "Microsoft.KeyVault",
            "keyVaultProperties": [
                {
                    "keyName": "bharrisonkey",
                    "keyVaultUri": "https://bharrisonkvtz6ium.vault.azure.net",
                    "keyVersion": "c90e6fcef73041d69801250f8bb60a86",
                    "identity": { <-- Encryption Identity block
                        "userAssignedIdentity": "/subscriptions/c6d2f82d-4baf-4b80-8d69-1fc6fbf92ad8/resourceGroups/bharrison-Tz6Ium/providers/Microsoft.ManagedIdentity/userAssignedIdentities/test"
                    }
                }
            ],
            "requireInfrastructureEncryption": false
        },
        "kafkaEnabled": true,
        "provisioningState": "Succeeded",
        "metricId": "c6d2f82d-4baf-4b80-8d69-1fc6fbf92ad8:bharrison-namespace-tz6ium",
        "createdAt": "2023-10-25T16:27:16.77Z",
        "updatedAt": "2023-10-25T16:34:43.68Z",
        "serviceBusEndpoint": "https://bharrison-namespace-Tz6Ium.servicebus.windows.net:443/",
        "status": "Active"
    }
}
```

Reply:
In which case we're modelling this elsewhere in the provider as a single (string) field (`user_assigned_identity_id`) - so I think we should be able to take that approach here?

Reply:
So ditch the ID block and just go with a string? 

I've got no issue with that, but I'll ask @katbyte to weigh in since she originally steered me towards the ID block.

I've got no preference, so either is fine with me.